src/rx-query.ts

```diff
@@ -648,19 +758,20 @@ function __ensureEqual<RxDocType>(rxQuery: RxQueryBase<RxDocType>): Promise<bool
                     // we got the new results, we do not have to re-execute, mustReExec stays false
                     ret = true; // true because results changed
                     rxQuery._setResultData(eventReduceResult.newResults as any);
+                    if (shouldUpdatePersistentQueryCache(rxQuery)) {
+                      await updatePersistentQueryCache(rxQuery);
```

Comment:
does this actually need to block `__ensureEqual`? i think it's safe to not...

Reply:
It doesn't, I just defaulted to the safest thing. FWIW I had no issues with writes blocking UI.

Reply:
Ah… I correct myself here, I actually had a pretty good reason to await at this point, it just doesn't make sense when persistence is done conditionally.